server/src/main/java/org/opensearch/index/shard/ReleasableRetryableRefreshListener.java

```diff
@@ -64,7 +64,13 @@ public ReleasableRetryableRefreshListener(ThreadPool threadPool) {
     @Override
     public final void afterRefresh(boolean didRefresh, CompositeEngine.ReleasableRef<CatalogSnapshot> catalogSnapshot) throws IOException {
         // TODO CompositeEngine filters CatalogSnapshotAwareListeners, keeping this for now
+
         afterRefresh(didRefresh);
+        try {
+            catalogSnapshot.close();
+        } catch (Exception ex) {
+
+        }
```

Comment:
Assuming this is the older version of the snapshot that would be dec referenced?